pypreprocess/reporting/pypreproc_reporter.py

```diff
@@ -0,0 +1,1001 @@
+                    "subject's %s") % (title_prefix, _brain_name))
+            log_link_text = embed_in_HTML('log_link_template.html',
+                                        for_substitution)
+            for_substitution['heading'] += ' '+log_link_text
+            if not only_native:
+                for_substitution['heading'] = ""
+                for_substitution['tooltip'] = ""
+
+            seg_plot_text = embed_in_HTML('report_sub_template.html',
+                                        for_substitution)
+            seg_plot.append(seg_plot_text)
+
+    if log:
+        return '\n'.join(seg_plot), log_text
+    else:
+        return '\n'.join(seg_plot)
+
+def generate_normalization_report(subject_data, output_dir, tooltip=None,
+                                log=True, report_path=None):
+    
+    """ 
+    Creates plots associated with normalization and returns them as SVG url 
+    embedded in HTML. Calls generate_segmentation_report and 
+    generate_registration_report. 
+    
+    Parameters
+    ----------
+     subject_data: `SubjectData` instance
+       object that encapsulates the date for the subject (should have fields
+       like func, anat, output_dir, etc.)
+
+    output_dir: string
+        directory containing all the output files
+    
+    report_path: string, optional (default None)
+        path to the report HTML file
+
+    """
+
+    _set_templates()
+    norm_plot = []
+    logs = []
+    subject_data._set_session_ids()
+
+    warped_tpms = dict(
+        (tpm, getattr(subject_data, tpm, None))
+        for tpm in ["mwgm", "mwwm", "mwcsf"])
+    segmented = warped_tpms.values().count(None) < len(warped_tpms)
+
+    if segmented:
+        plot_text = generate_segmentation_report(
+                subject_data=subject_data,
+                output_dir=subject_data.output_dir,
+                subject_gm_file=warped_tpms["mwgm"],
+                subject_wm_file=warped_tpms["mwwm"],
+                subject_csf_file=warped_tpms["mwcsf"],
+                comment="warped",log=False, 
+                report_path=report_path)
+        norm_plot.append(plot_text)
+        
+    for brain_name, brain, cmap in zip(
+            ['anatomical_image', 'mean_functional_image'],
+            [subject_data.anat, subject_data.func],
+            [cm.gray, cm.nipy_spectral]):
+        if not brain:
+            continue
+
+        if log:
+            for_substitution = {}
+            for_substitution['heading'] = "Normalization of %s" % brain_name
+            for_substitution['id_link'] = for_substitution['heading'].replace(
+                                                                " ", "_")
+            for_substitution['log'] = get_log_text(brain)
+            for_substitution['log_link'] = 'file:///'+os.path.join(output_dir
+                                            , 'nilearn_report_log.html#'
+                                            )+for_substitution['id_link']
```

Comment:
not very nice nor pep8...
maybe you could define and store the path in a temporary variable, that makes the code more readable.